application/src/main/java/org/togetherjava/tjbot/features/basic/ApplicationCreateCommand.java

```diff
@@ -0,0 +1,411 @@
+     * for.
+     *
+     * @param event the modal interaction event triggering the application submission
+     * @param args the arguments provided in the application submission
+     * @param answer the answer provided by the applicant to the default question
+     */
+    private void sendApplicationResult(final ModalInteractionEvent event, List<String> args,
+            String answer) {
+        Guild guild = event.getGuild();
+        if (args.size() != 2 || guild == null) {
+            return;
+        }
+
+        Optional<TextChannel> applicationChannel = getApplicationChannel(guild);
+        if (applicationChannel.isEmpty()) {
+            return;
+        }
+
+        User applicant = event.getUser();
+        EmbedBuilder embed =
+                new EmbedBuilder().setAuthor(applicant.getName(), null, applicant.getAvatarUrl())
+                    .setColor(AMBIENT_COLOR)
+                    .setTimestamp(Instant.now())
+                    .setFooter("Submitted at");
+
+        String roleString = args.getLast();
+        MessageEmbed.Field roleField = new MessageEmbed.Field("Role", roleString, false);
+        embed.addField(roleField);
+
+        MessageEmbed.Field answerField = new MessageEmbed.Field(DEFAULT_QUESTION, answer, false);
+        embed.addField(answerField);
+
+        applicationChannel.get().sendMessageEmbeds(embed.build()).queue();
+    }
+
+    /**
+     * Sends the initial embed and a button which displays role openings.
+     *
+     * @param event the command interaction event triggering the menu
+     */
+    private void sendMenu(final CommandInteraction event) {
+        MessageEmbed embed = createApplicationEmbed();
+
+        StringSelectMenu.Builder menuBuilder = StringSelectMenu
+            .create(generateComponentId(Lifespan.PERMANENT, event.getUser().getId()))
+            .setPlaceholder("Select role to apply for")
+            .setRequiredRange(1, 1);
+
+        addRolesToMenu(menuBuilder, event.getOptions());
+
+        event.replyEmbeds(embed).addActionRow(menuBuilder.build()).queue();
+    }
+
+    private static MessageEmbed createApplicationEmbed() {
+        return new EmbedBuilder().setTitle("Apply for roles")
+            .setDescription(
+                    """
+                            We are always looking for community members that want to contribute to our community \
+                            and take charge. If you are interested, you can apply for various positions here!""")
+            .setColor(AMBIENT_COLOR)
+            .build();
+    }
+
+    /**
+     * Wrapper class which represents a menu role for the application create command.
+     * <p>
+     * The reason this exists is due to the fact that {@link StringSelectMenu.Builder} does not have
+     * a method which takes emojis as input as of writing this, so we have to elegantly pass in
+     * custom data from this POJO.
+     */
+    private static class MenuRole {
```

Comment:
This can be a `record`

Reply:
`MenuRole` can not be a record the way it is currently used is due to the fact that its fields are set based on a few conditions located in `addRolesToMenu()` that's found in the same file. Moreover, there wouldn't be a lot to gain by converting it into a record, unless there's something I have overlooked.

Reply:
You could make it a `record` and put the conditions into a custom constructor

Reply:
Worked with Chris, came to a good solution.